fluentbit-kafka-fluentd/README.md

```diff
@@ -0,0 +1,175 @@
+          client_cert '/var/run/ocp-collector/secrets/fluentd/tls.crt'
+          ca_file '/var/run/ocp-collector/secrets/fluentd/ca-bundle.crt'
+          type_name _doc
+        </match>
+
+        <match *>
+          @type elasticsearch
+          host elasticsearch.openshift-logging.svc.cluster.local
+          port 9200
+          scheme https
+          ssl_version TLSv1_2
+          index_name catchall
+          user fluentd
+          password changeme
+
+          client_key '/var/run/ocp-collector/secrets/fluentd/tls.key'
+          client_cert '/var/run/ocp-collector/secrets/fluentd/tls.crt'
+          ca_file '/var/run/ocp-collector/secrets/fluentd/ca-bundle.crt'
+          type_name _doc
+        </match>
+    ```
+1. Create fluentd deployment:
+    ```
+    oc create -f fluentd-deployment.yaml
+    ```
+1. create fluentbit cm:
+    ```
+    oc create configmap fluentbit --from-file=fluent-bit.conf=fluent-bit.conf --from-file=parsers.conf=parsers.conf
+    ```
+1. create fluentbit ds:
+    ```
+    oc create -f fluentbit.yaml
+    ```
+
+## Current Gaps
+
+Currently the Fluentd k8s metadata plugin only correctly parses the container information from the tag which is based on the file path. There is an option to pull from a nested "kubernetes" field on the record, however this fails when trying to parse out the timestamp in the plugin. We would need to either fix this, or allow us to specify a field to parse out in lieu of using the tag in order to collect metadata at the aggregator level.
+
+// the issue is that fluentd in-kafka automatically parses the time field to be a float instead of giving us an option to keep it as a string -- this would let us get around this error
+
+As a more immediate work around we could keep collecting metadata at the collector level (in this case, Fluentbit [5]), however there would be less calls to the kubernetes service if this was at the aggregator level since we can independently scale Fluentd for aggregation separate from the number of nodes we are collecting from.
+
+## Next Steps and Recommendations
+
+There seemed to be a good performance increase over our current implementation of Fluentd -> Elasticsearch. I think some next steps would be to address the gap of the fabric8.io metadata plugin mentioned above, and have our performance team do a capacity test to figure out:
+1. how well can fluentbit keep up with log ingestion for very noisy logs?
+1. how many fluentd do we need per x rate of logs?
+1. at what point do we need to horizontally scale our kafka pods?
+1. verify/tune rate of log aggregation to prevent duplicate logs with multiple topic consumers
+
+I believe we should eventually move to make this our default configuration. As long as a [new] consumer can send logs to `kafka` then we should be able to configure `fluentd` to write to different endpoints. I can see this being the future state of log forwarding as well.
```

Comment:
Tangential to this effort but I think its fair to evaluate if we desire to ultimately deploy kafka as the messing solution.  AFAIK its a java app and poses the same issues we currently have with Elasticsearch: it potentially requires large amounts of RAM to support infrastructure which has been considered undesirable.  I presume it possible for us to swap in something like MRG (or whatever its called now) to accomplish the same thing for less cost with regards to node resources.

Reply:
as long as we have collectors that can write to it, and fluentd can read from it we could use a different messaging queue solution. currently kafka seems to be the most widely used one

Reply:
What's MRG?